apps/website/netlify/edge-functions/strangler.ts

```diff
@@ -0,0 +1,85 @@
+import { Config, Context } from 'https://edge.netlify.com';
+
+type LegacyHost = {
+  url: string;
+  urlFilter?: (url: URL) => boolean;
+  responseFilter?: (response: Response) => boolean;
+};
+
+const legacySystems: Array<LegacyHost> = [
+  {
+    // For Standard drupal redirects, we are interested in any
+    // url (therefore no urlFilter) and only in redirects (301, 302).
+    url: Deno.env.get('DRUPAL_EXTERNAL_URL') || 'http://localhost:8888',
+    responseFilter: (response) => [301, 302].includes(response.status),
+  },
+];
+
+export default async function strangler(
+  originalRequest: Request,
+  context: Context,
+) {
+  // Test if netlify can handle the request.
+  // Clone the request, or the invocation of next() will consume the body.
+  const netlifyResult = await context.next(originalRequest.clone(), {
+    sendConditionalRequest: false,
+  });
+
+  if (netlifyResult.status !== 404) {
+    return netlifyResult;
+  }
+
+  // Otherwise, pass the request to the legacy applications.
+  for (const legacyUrl of legacySystems) {
+    const request = originalRequest.clone();
+    const targetUrl = new URL(legacyUrl.url);
+    // Check if we even want to proxy this request.
+    // Skip if the urlFilter exists and returns false.
+    if (legacyUrl.urlFilter && !legacyUrl.urlFilter(targetUrl)) {
+      console.log('Skipping', legacyUrl.url);
```

Comment:
Will we see this in the Publisher logs? If so, maybe let's prefix it with `[strangler] `? (Same for other `console.log`s)

Reply:
But if it's visible in the log, that can be very good help for debugging

Reply:
But its also a lot of logs.

Reply:
True, let's remove them